src/main/java/de/tum/in/www1/artemis/service/FileService.java

```diff
@@ -1035,9 +1039,23 @@ public Path getUniquePath(String path) {
                 log.warn("could not create the directories for the path {}", uniquePath);
             }
         }
+
         return uniquePath;
     }
 
+    /**
+     * create a unique path by appending a folder named with the current milliseconds (e.g. 1609579674868) of the system and schedules it for deletion
+     *
+     * @param path                 the original path, e.g. /opt/artemis/repos-download
+     * @param deleteDelayInMinutes the delay in minutes after which the path should be deleted
+     * @return the unique path, e.g. /opt/artemis/repos-download/1609579674868
+     */
+    public Path getTemporaryUniquePath(String path, long deleteDelayInMinutes) {
```

Comment:
Please don’t introduce new methods to the file service that handle paths as `String`. At least the parameters for `public` methods should be proper `Path`s, it of course might be necessary to convert internally `toString` to be able to call other methods of the `FileService` until they are properly refactored.